cdap-cli/src/main/java/co/cask/cdap/cli/command/SetPreferencesCommand.java

```diff
@@ -47,19 +48,26 @@ public void printSuccessMessage(PrintStream printStream, ElementType type) {
 
   @Override
   public void perform(Arguments arguments, PrintStream printStream) throws Exception {
-    String runtimeArgs = arguments.get(ArgumentName.RUNTIME_ARGS.toString());
-    Map<String, String> args = ArgumentParser.parseMap(runtimeArgs, ArgumentName.RUNTIME_ARGS.toString());
-    setPreferences(arguments, printStream, args);
+    String[] programIdParts = new String[0];
+    if (arguments.hasArgument(type.getArgumentName().toString())) {
+      programIdParts = arguments.get(type.getArgumentName().toString()).split("\\.");
+      if (programIdParts.length < 2) {
```

Comment:
how come the command is parsing itself and checking the length instead of calling setPreferences(arguments, printStream, args)? Doesn't it mean it won't work for instance, namespace, and app preferences?